File: client/lib/Model/Caregiver.dart
import 'Learner.dart';
import 'User.dart';

class Caregiver extends User {
  final String caregiverID;
  late List<Learner> learners;
  Caregiver(
      {required this.caregiverID,
      required super.firstName,
      required super.lastName,
      required super.email,
      required super.password,
      required this.learners});

  factory Caregiver.fromJson(Map<String, dynamic> json) {
    return Caregiver(
        caregiverID: json['caregiverID'],
        firstName: json['firstName'],
        lastName: json['lastName'],
        email: json['email'],
        password: json['password'],
        learners: json['learners']);
  }

  String getCaregiverID() {
    return this.caregiverID;
  }

  List<Learner> getLearners() {
    return this.learners;
  }
}

File: server/Model/Caregiver.js
import "package:myplanner_nodeapp/server/Model/Learner.js";

const mongoose = require("mongoose");
const Schema = mongoose.Schema;
const caregiverSchema = new mongoose.Schema({
  firstName: {
    type: String,
    required: true,
  },
  lastName: {
    type: String,
    required: true,
  },
  email: {
    unique: true,
    type: String,
    required: true,
  },
  password: {
    type: String,
    required: true,
  },
  learners: [{ type: Schema.Types.ObjectId, ref: "learner" }],
});
module.exports = mongoose.model("caregiver", caregiverSchema); //node's way of exporting a file  (name of table, the actual table made)
